cli/pkg/scanners/file_scanner.go

```diff
@@ -29,9 +30,15 @@ func (f *FileScanner) Scan() ([]pkg.Earthfile, error) {
 }
 
 func (f *FileScanner) ScanForTarget(target string) ([]pkg.Earthfile, error) {
+	// Should start with given target.
+	// Allowing an optional hyphen followed by one or more lowercase letters or numbers
+	regexPattern := "^" + regexp.QuoteMeta(target) + "(?:-[a-z0-9]+)?$"
+
+	r, _ := regexp.Compile(regexPattern)
+
 	earthfiles, err := f.scan(func(e pkg.Earthfile) (bool, error) {
 		for _, t := range e.Targets {
-			if t.Name == target {
+			if r.MatchString(t.Name) {
```

Comment:
So the issue here is this will change the current behavior of the CLI. As it stands, passing `-t target` will _only_ return targets with the exact name of `target`. With this change, the CLI will _also_ return `target-*` targets. 

I typically try to avoid making backward incompatible changes unless they are really necessary. It's frustrating for a flag to suddenly change its logic fairly dramatically. Instead, we should force the end-user to explicitly specify when they intend to glob targets. For example:

```
-t target -t target-*
```

Would match `target` and `target-test`.

```
-t target
```

Would _only_ match `target` and nothing else.